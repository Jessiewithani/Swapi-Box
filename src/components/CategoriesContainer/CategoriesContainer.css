h1 {
    color: white;
    font-size: 75px;
}

.bow {
    padding: 25px 90px;
}

.categoriesContainer {
    border-radius: 10px;
    display: grid;
    grid-template-columns: 210px 210px 210px 210px;
    grid-gap: 20px;
    height: 70%;
    justify-content: flex-end;
    overflow: scroll;
    padding: 50px;
}

.helloHeader {
    background-color: pink;
    border-radius: 10px;
    display: flex;
    height: 50px;
}

h3 {
    font-size: 30px;
    margin: 12px;
}

.pinkbow {
    width: 300px;
    position: fixed;
    right: 30px;
    top: -15px;
}

.fav-heart {
    width: 20px;
    margin-right: 0px;
    
}

.pick-favorites {
    margin-right: 16.5em;
}