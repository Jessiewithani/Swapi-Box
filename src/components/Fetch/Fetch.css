.navContainer {
  width: 20%;
  justify-content: flex-start;
}

.buttonImages {
  height: 130px;
  width: 130px;
  margin: 1px 75px;
}

button {
  border-radius: 8px;
  display: block;
  font-family: 'Dunkin';
  color: #FF0075;
  font-size: 18px;
  height: 30px;
  margin: 5px auto;
  width: 70%;
}

button:hover {
  color: #72C676;
}