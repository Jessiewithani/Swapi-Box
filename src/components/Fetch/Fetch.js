import React, {Component} from 'react'

class Fetch extends Component {
  constructor(props) {
    super(props)
    this.state = {
      people: [],
      vehicles : [],
      planets: []
    }
  }

  componentDidMount() {
    const people = fetch('https://swapi.co/api/people/').then(response => response.json());
    const planets = fetch('https://swapi.co/api/planets/').then(response => response.json());
    const vehicles = fetch('https://swapi.co/api/vehicles/').then(response => response.json());
    const combined = { 'people': [], 'planets': [], 'vehicles': [] }

    Promise.all([people, planets, vehicles])
      .then(function (values) {
        combined['people'] = values[0];
        combined['planets'] = values[1];
        combined['vehicles'] = values[2];
        return combined;
      })
      .then(combined => this.setState({
        people: combined.people.results,
        vehicles: combined.vehicles.results,
        planets: combined.planets.results
      }))
  }

  render() {
    return (
      <div>{this.state.people.map(person => <h3>{person.name}</h3>)}</div>
    )
  }
}

export default Fetch;