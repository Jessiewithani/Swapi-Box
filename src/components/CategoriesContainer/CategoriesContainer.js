import React from 'react';
import '../CategoriesContainer/CategoriesContainer.css';
import CategoryCard from '../CategoryCard/CategoryCard';
import PropTypes from 'prop-types';

const CategoriesContainer = ({ data, toggleFavorite, countFavorites }) => {
  const displayCards = data.map(card => <CategoryCard 
    key={card.name + Date.now()}
    type={card.type}
    name={card.name}
    homeworld= {card.homeworld}
    species={card.species}
    population={card.population}
    model={card.model}
    vehicle_class={card.vehicle_class}
    passengers={card.passengers}
    terrain={card.terrain}
    climate={card.climate}
    residents={card.resident}
    toggleFavorite={toggleFavorite}
    />);

    return (
      <div className="bow">
        <h1>Hello Star Wars</h1>
        <header className="helloHeader">
          <h3>Faves: {countFavorites()}</h3>
          {(countFavorites() <1) && <h3>Add Favorites!</h3>}
          <h3>Image</h3>
        </header>
        <article className= "categoriesContainer">
          {displayCards}
        </article>
      </div>
    )
}

export default CategoriesContainer;

CategoriesContainer.propTypes = {
  data: PropTypes.arrayOf(PropTypes.object),
  toggleFavorite: PropTypes.func
}