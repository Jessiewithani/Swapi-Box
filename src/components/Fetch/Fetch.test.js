import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router';
import Fetch from './Fetch';
import CategoriesContainer from '../CategoriesContainer/CategoriesContainer';
import NoMatch from '../NoMatch/NoMatch'

  describe('Fetch routes using memory router', () => {
    it('should show Fetch component for / router (using memory router)', () => {
      const component = mount(<MemoryRouter initialEntries={['/']} >
        <Fetch checkLoading={jest.fn()}/>
      </MemoryRouter>
      );
      expect(component.find(Fetch)).toHaveLength(1);
    });

    it('should show a CategoriesContainer component for /people router', () => {
      const mockPeopleData = [{ name: 'Luke Skywalker' }, { name: 'Darth Vader' }]
      const component = mount(<MemoryRouter initialEntries={['/people']}>
        <CategoriesContainer countFavorites={jest.fn()} data={mockPeopleData}/>
      </MemoryRouter>)
      expect(component.find(CategoriesContainer)).toHaveLength(1);
    });

    it('should show a CategoriesContainer component for /vehicles router', () => {
      const mockVehicleData = [{ name: 'Sand Crawler' }, { name: 'Snowspeeder' }]
      const component = mount(<MemoryRouter initialEntries={['/people']}>
        <CategoriesContainer countFavorites={jest.fn()} data={mockVehicleData} />
      </MemoryRouter>)
      expect(component.find(CategoriesContainer)).toHaveLength(1);
    });

    it('should show a CategoriesContainer component for /planets router', () => {
      const mockPlanetData = [{ name: 'Hoth' }, { name: 'Endor' }]
      const component = mount(<MemoryRouter initialEntries={['/people']}>
        <CategoriesContainer countFavorites={jest.fn()} data={mockPlanetData} />
      </MemoryRouter>)
      expect(component.find(CategoriesContainer)).toHaveLength(1);
    });

    it('should show a CategoriesContainer component for /favorites router', () => {
      const mockFavoritesData = [{ name: 'Hoth' }, { name: 'Luke Skywalker' }]
      const component = mount(<MemoryRouter initialEntries={['/people']}>
        <CategoriesContainer countFavorites={jest.fn()} data={mockFavoritesData} />
      </MemoryRouter>)
      expect(component.find(CategoriesContainer)).toHaveLength(1);
    });
    it('should show a NoMatch component for route not defined', () => {
      const component = mount(<MemoryRouter initialEntries={['/abc']}>
        <Fetch checkLoading={jest.fn()}/>
      </MemoryRouter>)
      expect(component.find(NoMatch)).toHaveLength(1);
    });

    it('should match snapshot', () => {
      const wrapper = shallow(<Fetch checkLoading={jest.fn()}/>)
      expect(wrapper).toMatchSnapshot();
    });

    it('should update state as favorites are toggled', () => {
      const wrapper = shallow(<Fetch checkLoading={jest.fn()}/>);
      const mockFavorite = { name: "Luke Skywalker", type: "people", homeworld: "Tatooine", species: "Human", population: '200000' };
      wrapper.setState({people: [mockFavorite]})
      const expected = [mockFavorite];
      wrapper.instance().toggleFavorite("Luke Skywalker", "people");
      expect(wrapper.state('favorites')).toEqual(expected);
    });
    

  });
