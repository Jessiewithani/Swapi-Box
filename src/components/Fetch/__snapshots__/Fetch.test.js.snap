// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fetch routes using memory router should match snapshot 1`] = `
<Fragment>
  <BrowserRouter>
    <nav
      className="navContainer"
    >
      <NavLink
        className="nav"
        to="/people"
      >
        <img
          alt="Princess Leia"
          className="buttonImages"
          src="PrincessLeia.png"
        />
        <button>
          PEOPLE
        </button>
      </NavLink>
      <NavLink
        className="nav"
        to="/vehicles"
      >
        <img
          alt="Chewbacca"
          className="buttonImages"
          src="Chewbacca.png"
        />
        <button>
          VEHICLES
        </button>
      </NavLink>
      <NavLink
        className="nav"
        to="/planets"
      >
        <img
          alt="DeathStar"
          className="buttonImages"
          src="DeathStar.png"
        />
        <button>
          PLANETS
        </button>
      </NavLink>
      <NavLink
        className="nav"
        to="/favorites"
      >
        <img
          alt="Yoda"
          className="buttonImages"
          src="Yoda.png"
        />
        <button>
          FAVORITES
        </button>
      </NavLink>
    </nav>
    <Switch>
      <Route
        exact={true}
        path="/people"
        render={[Function]}
      />
      <Route
        path="/vehicles"
        render={[Function]}
      />
      <Route
        path="/planets"
        render={[Function]}
      />
      <Route
        path="/favorites"
        render={[Function]}
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </BrowserRouter>
</Fragment>
`;
